### DISCUSSION TITLE
Vb.net -  Image(Of Rgba32)  not working

### DISCUSSION THREAD

**Discussion by fmonteiro-pt** [SixLabors/ImageSharp#2626] [Q&A]
Hi, I would like to know how to solve this example.
 Dim image As Image(Of Rgba32) , does not seem valid in Visual Studio - Visual Basic

` 
Dim base64ImageData As String = "iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAIAAACQkWg2AAAAWElEQVR42mP8z8AAROP8TT10JAAAAAElFTkSuQmCC"
 base64ImageData = base64ImageData.Replace(" ", "").Replace(vbCrLf, "").Replace(vbLf, "").Replace(vbCr, "")
 Dim imageBytes As Byte() = Convert.FromBase64String(base64ImageData)
 Using imageStream As New MemoryStream(imageBytes)
     Dim image As Image(Of Rgba32) = image.Load(Of Rgba32)(imageStream)
 End Using
`

**Comment by tocsoft** [comment]
The `Load()` method your trying to call is a static method on the `Image` class, it looks like your trying to call in on the local `image` instance you have just declared. Try changing 
```diff
-     Dim image As Image(Of Rgba32) = image.Load(Of Rgba32)(imageStream)
+     Dim image As Image(Of Rgba32) = Image.Load(Of Rgba32)(imageStream)
```
note the capital `I` in the `Image.Load(` call

big caveat here is that I'm not a VB developer and kind of just guessing as you failed to actually provide any real detail on the error your actually seeing.

PS. You really need to make sure you do a few things things when seeking help on an open source projects repo, you should provide all the error information you are seeing, just saying  'doesn't seem valid' isn't anywhere near as helpful as trying to compile and then reporting that the compiler return 'XXX' error message on line Y... also when providing code samples please to to make sure they are formatted correctly, preferable with colorized style applied. If you don't know how to do that you want to read the markdown docs linked below every comment box on github.

**Reply by fmonteiro-pt** [reply]
Thanks a lot for the try, but that is not it...

![image](https://github.com/SixLabors/ImageSharp/assets/47044054/e9bdb97c-c825-4775-8276-4c214c984f4a)

**Reply by antonfirsov** [reply]
I think there was a typo in the reply:

```diff
-     Dim image As Image(Of Rgba32) = image.Load(Of Rgba32)(imageStream)
+     Dim image As Image = Image.Load(imageStream)
```

**Comment by fmonteiro-pt** [comment]
It works like this:

`
  Dim base64ImageData As String = "iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAA5SURBVDhPpcixDQAgDMCw/v904QAPRETykrntJ86Cs+AsOAvOgrPgLDgLzoKz4Cw4C86Cs+B8NHsAm8z/AS15iaAAAAAASUVORK5CYII="
  Dim imageBytes As Byte() = Convert.FromBase64String(base64ImageData)
  Using imageStream As New IO.MemoryStream(imageBytes)
      Dim pic = Image.Load(imageStream)
  End Using
`

What is the problem with the "(of Rgba32)"??

**Reply by antonfirsov** [reply]
> What is the problem with the "(of Rgba32)"??

See https://github.com/dotnet/vblang/issues/300. Based on the activity on the issues like that one, I think it's safe to state that VB is a maintenance-only legacy language.

**Comment by tiesont** [comment]
I assume you're running into something related to this: https://github.com/dotnet/vblang/issues/297

Basically, VB.NET is not compatible with some of the types being used in ImageSharp. This is just a guess, though, as I rarely code in VB.
